src/sage/crypto/boolean_function.pyx

```diff
@@ -1341,6 +1368,7 @@ cdef class BooleanFunction(SageObject):
         """
         self._clear_cache()
         bitset_set_to(self._truth_table, int(i), int(y)&1)
+        self._hamming_weight = -1
```

Comment:
Is it possible to update the hamming weight directly by either +1 or -1 (if it's computed already)? Something like

```python
if self._hamming_weight != -1 and self._truth_table[int(i)] != int(y) & 1:
    if int(y) & 1: self._hamming_weight += 1
    else: self._hamming_weight -= 1
```
or something